qldpc/abstract.py

```diff
@@ -227,6 +236,70 @@ def from_generators(
         """Construct a group from generators."""
         return Group(comb.PermutationGroup(*generators), field, lift)
 
+    @classmethod
+    def from_permutation_mats(
+        cls, generators: Iterable[galois.FieldArray], space: list[bytes], field: int | None = None
+    ) -> Group:
+        """Constructs a Group from a given set of generating permutation matrices."""
+        finite_field = galois.GF(field or DEFAULT_FIELD_ORDER)
+        group_perms = []
+        for member in generators:
+            string = list(range(len(space)))
+            for index in range(len(space)):
+                current_vector = finite_field(np.frombuffer(space[index], dtype=np.uint8))
+                next_vector = member @ current_vector
+                next_index = space.index(next_vector.tobytes())
+                string[index] = next_index
+            group_perms.append(comb.Permutation(string))
+        return Group(comb.PermutationGroup(group_perms), field=field)
```

Comment:
I don't quite understand what this method is doing.  Let's discuss offline.

Reply:
Oh, it occurs to me that perhaps this method may not truly belong in `Groups`...  I guess that will get sorted out when we talk about it.

Reply:
Okay looking at it more closely I think I get what this is doing now.  As is, it does not really belong in `Group`, so maybe I'll just move it into `SpecialLinearGroup`.

Reply:
The input matrices we use here are not actually permutations.  The "from_permutation_mats" name was misleading.

Reply:
A bigger problem is https://github.com/Infleqtion/qLDPC/pull/15#discussion_r1480377183